src/app/accounts/update-account-custom-fields.ts

```diff
@@ -0,0 +1,48 @@
+import { getAccountsConfig } from "@config"
+
+import { CouldNotFindError } from "@domain/errors"
+import { NoAccountCustomFieldsError } from "@domain/accounts"
+
+import { AccountCustomFieldsRepository, AccountsRepository } from "@services/mongoose"
+
+const { customFields: customFieldsSchema } = getAccountsConfig()
+
+const defaultValues = customFieldsSchema.reduce((acc, val) => {
+  if (val.defaultValue !== undefined) {
+    acc[val.name] = val.defaultValue
+  }
+  return acc
+}, {} as { [k: string]: AccountCustomFieldValues })
+
+export const updateAccountCustomFields = async ({
+  accountId,
+  createdByUserId,
+  customFields,
+}: {
+  accountId: AccountId
+  createdByUserId: UserId
+  customFields: { [k: string]: AccountCustomFieldValues }
+}): Promise<AccountCustomFields | ApplicationError> => {
+  if (!customFieldsSchema || customFieldsSchema.length <= 0)
+    return new NoAccountCustomFieldsError()
+
+  const accountCustomFieldsRepo = AccountCustomFieldsRepository()
+
+  const account = await AccountsRepository().findById(accountId)
+  if (account instanceof Error) return account
+
+  const accountCustomFields = await accountCustomFieldsRepo.findById(account.id)
+  const isEmpty = accountCustomFields instanceof CouldNotFindError
+  if (accountCustomFields instanceof Error && !isEmpty) return accountCustomFields
+
+  let data = customFields
+  if (!isEmpty) {
+    data = { ...accountCustomFields.customFields, ...customFields }
+  }
+
+  return accountCustomFieldsRepo.persistNew({
```

Comment:
why are you using `repo.persistNew` in a use case called `update`?
Repos should expose `CRUD` methods - don't use `persistNew` (the `C` method) for an `update`.

Reply:
Updated https://github.com/GaloyMoney/galoy/pull/1465/commits/22c9f29082861a0a3ae06bfd1c0c82247e53368b